addon/components/polaris-data-table.js

```diff
@@ -1,525 +1,490 @@
+      collapsed = table.scrollWidth > scrollContainer.clientWidth;
+      scrollContainer.scrollLeft = 0;
+    }
+
+    this.setProperties(
+      assign(
+        {
+          collapsed,
+          heights: [],
+        },
+        this.calculateColumnVisibilityData(collapsed)
+      )
+    );
+
+    if (footerContent || !truncate) {
+      scheduleOnce('afterRender', this, this.setHeightsAndScrollPosition);
+    }
+  }
 
-        if (!truncate) {
-          return (heights = rows.map((row) => {
-            let fixedCell = row.hasChildNodes() && row.children[0];
-            return Math.max(row.clientHeight, fixedCell.clientHeight);
-          }));
-        }
+  scrollListener() {
+    if (this.get('isDestroying') || this.get('isDestroyed')) {
+      return;
+    }
 
-        if (footerContent) {
-          let footerCellHeight =
-            rows[rows.length - 1].childNodes[0].clientHeight;
-          heights = [footerCellHeight];
-        }
+    this.setProperties(
+      this.calculateColumnVisibilityData(this.get('collapsed'))
+    );
+  }
 
-        return heights;
-      }
-    },
+  tallestCellHeights() {
+    let { footerContent, truncate, heights, table } = this;
 
-    addEventHandlers() {
-      this.addEventListener(window, 'resize', this.handleResize);
-      this.addEventListener(window, 'scroll', this.scrollListener, {
-        capture: true,
-      });
-    },
+    if (table) {
+      let rows = Array.from(table.getElementsByTagName('tr'));
 
-    init() {
-      this._super(...arguments);
+      if (!truncate) {
+        return (heights = rows.map((row) => {
+          let fixedCell = row.hasChildNodes() && row.children[0];
+          return Math.max(row.clientHeight, fixedCell.clientHeight);
+        }));
+      }
 
-      this.setProperties({
-        columnVisibilityData: [],
-        heights: [],
-        preservedScrollPosition: {},
-      });
-    },
+      if (footerContent) {
+        let footerCellHeight = rows[rows.length - 1].childNodes[0].clientHeight;
+        heights = [footerCellHeight];
+      }
 
-    didInsertElement() {
-      this._super(...arguments);
+      return heights;
+    }
+  }
 
-      this.handleResize();
+  addEventHandlers() {
+    this.addEventListener(window, 'resize', this.handleResize);
+    this.addEventListener(window, 'scroll', this.scrollListener, {
+      capture: true,
+    });
+  }
 
-      this.addEventHandlers();
-    },
+  init() {
+    super.init(...arguments);
 
-    didUpdateAttrs() {
-      this._super(...arguments);
+    this.setProperties({
+      columnVisibilityData: [],
+      heights: [],
+      preservedScrollPosition: {},
+    });
+  }
 
-      if (isEqual(this.get('oldAttrs'), this.get('attrs'))) {
-        return;
-      }
+  @action
+  dataTableDidInsert(element) {
+    this.set('dataTableElement', element);
+    this.handleResize();
+    this.addEventHandlers();
+  }
 
-      this.handleResize();
+  @action
+  dataTableDidUpdate() {
```

Comment:
same re naming tweak

we need to be careful here and likely check Polaris in what circumstances they do these updates...because currently this action will only fire once (we'll have to pass the properties that might change and need to re-trigger this action in the template_

Reply:
@vladucu looks like the only _public_ properties that get used in the `componentDidUpdate` method in the 3.10 DataTable are `footerContent` and `truncate`, so we can probably just pass those two properties into `did-update`.

The React version `return`s inside ``componentDidUpdate`` if `isEqual(prevProps, this.props)`, so I'm guessing we don't need to worry about any of our privately set properties needing to trigger an update.